src/tests/banners/bannerSelection.tsx

```diff
@@ -34,70 +49,150 @@ const fetchBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date>
+            fiveMinutes,
+            'fetchEngagementBannerDeployTime_united-kingdom',
+        ),
+        'united-states': cacheAsync(
+            fetchBannerDeployTime('united-states', 'contributions'),
+            fiveMinutes,
+            'fetchEngagementBannerDeployTime_united-states',
+        ),
+        australia: cacheAsync(
+            fetchBannerDeployTime('australia', 'contributions'),
+            fiveMinutes,
+            'fetchEngagementBannerDeployTime_australia',
+        ),
+        'rest-of-world': cacheAsync(
+            fetchBannerDeployTime('rest-of-world', 'contributions'),
+            fiveMinutes,
+            'fetchEngagementBannerDeployTime_rest-of-world',
+        ),
+        // Contributions doesn't separate europe from row
+        'european-union': cacheAsync(
+            fetchBannerDeployTime('rest-of-world', 'contributions'),
+            fiveMinutes,
+            'fetchEngagementBannerDeployTime_rest-of-world',
+        ),
+    },
+    subscriptions: {
+        'united-kingdom': cacheAsync(
+            fetchBannerDeployTime('united-kingdom', 'subscriptions'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_united-kingdom',
+        ),
+        'united-states': cacheAsync(
+            fetchBannerDeployTime('united-states', 'subscriptions'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_united-states',
+        ),
+        australia: cacheAsync(
+            fetchBannerDeployTime('australia', 'subscriptions'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_australia',
+        ),
+        'rest-of-world': cacheAsync(
+            fetchBannerDeployTime('rest-of-world', 'subscriptions'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_rest-of-world',
+        ),
+        // Subscriptions separates europe from row
+        'european-union': cacheAsync(
+            fetchBannerDeployTime('european-union', 'subscriptions'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_european-union',
+        ),
+    },
 };
 
 // Has the banner been redeployed since the user last closed it?
-const redeployedSinceLastClosed = (targeting: BannerTargeting): Promise<boolean> => {
-    if (targeting.engagementBannerLastClosedAt) {
-        const closedAt = targeting.engagementBannerLastClosedAt;
-        const region = readerRevenueRegionFromCountryCode(targeting.countryCode);
-
-        const [, getCached] = caches[region];
+export const redeployedSinceLastClosed = (
+    targeting: BannerTargeting,
+    bannerType: BannerType,
+): Promise<boolean> => {
+    const { subscriptionsBannerLastClosedAt, engagementBannerLastClosedAt } = targeting;
 
-        return getCached().then(deployDate => deployDate > new Date(closedAt));
-    } else {
+    if (
+        (bannerType === 'subscriptions' &&
+            typeof subscriptionsBannerLastClosedAt === 'undefined') ||
```

Comment:
We might be able to replace these `typeof subscriptionsBannerLastClosedAt === 'undefined'` checks with falsy checks now. eg. `(bannerType === 'subscriptions' && !subscriptionsBannerLastClosedAt)`. I think the `typeof` check was only introduced in attempt to resolve a TypeScript issue, which it failed to do.

Reply:
👍